custom_components/tuya_local/devices/ferroli_titano_twin.yaml

```diff
@@ -0,0 +1,107 @@
+name: Ferroli Titano Twin
+products:
+  - id: lci0ebsz7ftaruaf
+    name: DT EWH
+primary_entity:
+  entity: water_heater
+  dps:
+    - id: 1
+      name: operation_mode
+      type: boolean
+      mapping:
+        - dps_val: false
+          value: "off"
+        - dps_val: true
+          constraint: work_mode
+          conditions:
+            - dps_val: manual
+              value: electric
+            - dps_val: eco
+              value: eco
+              icon: "mdi:leaf"
+    - id: 2
+      type: string
+      name: work_mode
+      hidden: true
+    - id: 9
+      type: integer
+      name: temperature
+      range:
+        min: 40
+        max: 80
+      mapping:
+        - step: 5
+      unit: C
+    - id: 10
+      type: integer
+      name: current_temperature
```

Comment:
I would leave the dp: 101 here to expose the raw fault_code, since the binary_sensor is mapping it to true or false, so the actual value is lost. 

It can be defined in both places at once - the binary sensor is easier to add to the UI, and check in an automation where the fault code is not important, but if you need to differentiate faults, the information will still be on an extra attribute here.